#### Created by: Sabrina Karam

## Heroes of Pymoli Data Analysis

- There are 780 players who are active in the game; out of the total players the majority are males taking up 84% of the population while female players take up approximately 14%.

- The peak age demographic would be between the ages of 20-24 years old (47%) and the other group would be teenages in between 15-19 years old (~18%).

In [1]:
#set up the dependencies
import pandas as pd
import numpy as np

In [2]:
#create a file path
heroes_path = "Resources/purchase_data.csv"

In [3]:
#read the file and create a table
purchase_game = pd.read_csv(heroes_path)
purchase_game_df = pd.DataFrame(purchase_game)
purchase_game_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [4]:
#determine the number of players
sum_of_players = purchase_game_df["SN"].count()
sum_of_players

780

### Purchase Analysis

- Ran basic calculations to determine the number of unique items, average price and other attributes to help create a DataFrame to hold the results


- Cleaned up DataFrame to a more organized formatting


- Display the DataFrame

In [5]:
#calculate total number of unique items
unique_items = len(purchase_game_df["Item ID"].unique())
unique_items

179

In [6]:
#calculate the total number of purchases
purchase_total = purchase_game_df["Purchase ID"].count()
purchase_total

780

In [7]:
#calculate the total revenue for game
total_revenue = purchase_game_df["Price"].sum()
total_revenue

2379.77

In [8]:
#calculate the average price
average_prices1 = purchase_game_df["Price"].mean()
average_prices1

3.050987179487176

In [9]:
#calculate the average price
average_prices1 = purchase_game_df["Price"].mean()
average_prices1

average_prices2 = total_revenue/purchase_total
average_prices2

#average_prices1 = average_prices2

3.0509871794871795

In [10]:
#create a summary for the data as a frame format
purchase_analysis_df = pd.DataFrame([{"Number of Unique Items": unique_items, "Average Prices" : average_prices1, "Number of Purchases" : purchase_total, "Total Revenue" : total_revenue}])
purchase_analysis_df["Average Prices"] = purchase_analysis_df["Average Prices"].map("${:,.2f}".format)
purchase_analysis_df["Total Revenue"] = purchase_analysis_df["Total Revenue"].map("${:,.2f}".format)
purchase_analysis_df

#reset the column order for the summary data frame
organize_purchase_analysis_df = purchase_analysis_df[["Number of Unique Items" , "Average Prices" , "Number of Purchases" , "Total Revenue"]]
organize_purchase_analysis_df

Number of Unique Items Average Prices  Number of Purchases Total Revenue
0                     179          $3.05                  780     $2,379.77

### GENDER DEMOGRAPHICS

- Ran calculations to help calculate the number of unique, average price, and other attributes help create a DataFrame to hold all results


- Percentage and Count of Male & Female Players


- Percentage and Count of Other/Non-Disclosed Genders


- Display the DataFrame

In [12]:
#calculate gender differences between players
gender_df = pd.DataFrame(purchase_game_df["Gender"].value_counts())
gender_df

gender_percentage = (purchase_game_df["Gender"].value_counts()/sum_of_players)*100
gender_percentage

#take the calculations and add them to the data frame as new column
gender_df["Percentage of Players"] = gender_percentage
gender_df["Percentage of Players"] = gender_df["Percentage of Players"].map("{:,.2f}%".format)
gender_df

#re-organize the order of the columns
organize_gender_df = gender_df[["Percentage of Players" , "Gender"]]
organize_gender_df

#rename the columns using the .rename command
final_gender_df = organize_gender_df.rename(columns={"Gender" : "Total Counts"})
final_gender_df

Percentage of Players  Total Counts
Male                                 83.59%           652
Female                               14.49%           113
Other / Non-Disclosed                 1.92%            15

In [13]:
#use the group by function to separate the data into fields with Gender values
gender_purchased_data_df = purchase_game_df.groupby(["Gender"])

#print(gender_purchcased_data_df) by using a function
gender_purchased_data_df["Purchase ID"].count().head(10)

Gender
Female                   113
Male                     652
Other / Non-Disclosed     15
Name: Purchase ID, dtype: int64

In [14]:
#calculate the total purchase value by gender
total_purchase_value = gender_purchased_data_df["Price"].sum()
total_purchase_value.head()
dlr_total_purchase_value = total_purchase_value.map("${:,.2f}".format)
dlr_total_purchase_value.head()

Gender
Female                     $361.94
Male                     $1,967.64
Other / Non-Disclosed       $50.19
Name: Price, dtype: object

In [15]:
#calculate the average purchase price by gender
average_purchase_price = gender_purchased_data_df["Price"].mean()
average_purchase_price.head()
dlr_average_purchase_price = average_purchase_price.map("${:,.2f}".format)
dlr_average_purchase_price.head()

Gender
Female                   $3.20
Male                     $3.02
Other / Non-Disclosed    $3.35
Name: Price, dtype: object

In [16]:
#total purchase value divided by purchase count by gender
normalized_totals = total_purchase_value/gender_purchased_data_df["Purchase ID"].count()
dlr_normalized_totals = normalized_totals.map("${:,.2f}".format)
dlr_normalized_totals.head()

Gender
Female                   $3.20
Male                     $3.02
Other / Non-Disclosed    $3.35
dtype: object

In [17]:
#organize the summary data for genders, and organize all columns in a Data Frame
organize_gender_purchase_data_df = pd.DataFrame(gender_purchased_data_df["Purchase ID"].count())
organize_gender_purchase_data_df["Average Purchase Price"] = dlr_average_purchase_price
organize_gender_purchase_data_df["Total Purchase Values"] = dlr_total_purchase_value
organize_gender_purchase_data_df["Normalized Totals"] = dlr_normalized_totals
organize_gender_purchase_data_df

Purchase ID Average Purchase Price  \
Gender                                                      
Female                         113                  $3.20   
Male                           652                  $3.02   
Other / Non-Disclosed           15                  $3.35   

                      Total Purchase Values Normalized Totals  
Gender                                                         
Female                              $361.94             $3.20  
Male                              $1,967.64             $3.02  
Other / Non-Disclosed                $50.19             $3.35

In [18]:
#calculate the summary of purchases DataFrame and group it by gender and rename columns
summary_gender_data_df = organize_gender_purchase_data_df.rename(columns={"Purchase ID": "Purchase Count"})
summary_gender_data_df

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Values Normalized Totals  
Gender                                                         
Female                              $361.94             $3.20  
Male                              $1,967.64             $3.02  
Other / Non-Disclosed                $50.19             $3.35

### AGE DEMOGRAPHIC ANALYSIS

- Created bins for the varying age brackets


- Calculate numbers and percentages by varying age groups


- Create the summary DataFrame to hold all results of calculations


- Display the Age Demographics as a table

In [19]:
#create bins where the purchase data will be located
age_bins = [0, 9.90, 14.90, 19.90, 24.90, 29.90, 34.90, 39.90, 99999]

In [20]:
#create names for the four bins
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]
group_by_age_purchase_data_df = purchase_game_df
group_by_age_purchase_data_df["Age Summary"] = pd.cut(group_by_age_purchase_data_df["Age"], age_bins, labels=group_names)
group_by_age_purchase_data_df

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price Age Summary  
0    Extraction, Quickblade Of Trembling Hands   3.53       20-24  
1                            Frenzied Scimitar   1.56         40+  
2                                 Final Critic   4.88       20-24  
3                                  Blindscythe   3.27       20-24  
4                                         Fury   1.44       20-24  
..                                         ...    ...         ...  
775                                       Wolf   3.54       20-24  
776                           Exiled Doomblade   1.63       20-24  
777      Celeste, Incarnation of the Corrupted   3.46       20-24  
778                               Final Critic   4.19         <10  
779                                       Dawn   4.60       20-24  

[780 rows x 8 columns]

In [21]:
#calculate a group using the bins
group_by_age_purchase_data_df = group_by_age_purchase_data_df.groupby("Age Summary")
group_by_age_purchase_data_df.count()

Purchase ID   SN  Age  Gender  Item ID  Item Name  Price
Age Summary                                                          
<10                   23   23   23      23       23         23     23
10-14                 28   28   28      28       28         28     28
15-19                136  136  136     136      136        136    136
20-24                365  365  365     365      365        365    365
25-29                101  101  101     101      101        101    101
30-34                 73   73   73      73       73         73     73
35-39                 41   41   41      41       41         41     41
40+                   13   13   13      13       13         13     13

In [22]:
#utilize the new DataFrame
age_summary_df = pd.DataFrame(group_by_age_purchase_data_df.count())
age_summary_df

Purchase ID   SN  Age  Gender  Item ID  Item Name  Price
Age Summary                                                          
<10                   23   23   23      23       23         23     23
10-14                 28   28   28      28       28         28     28
15-19                136  136  136     136      136        136    136
20-24                365  365  365     365      365        365    365
25-29                101  101  101     101      101        101    101
30-34                 73   73   73      73       73         73     73
35-39                 41   41   41      41       41         41     41
40+                   13   13   13      13       13         13     13

In [24]:
#calculations on columns of the summary DataFrame
age_summary_df["Purchase ID"] = (age_summary_df["Purchase ID"]/sum_of_players)*100
age_summary_df

Purchase ID   SN  Age  Gender  Item ID  Item Name  Price
Age Summary                                                          
<10             2.948718   23   23      23       23         23     23
10-14           3.589744   28   28      28       28         28     28
15-19          17.435897  136  136     136      136        136    136
20-24          46.794872  365  365     365      365        365    365
25-29          12.948718  101  101     101      101        101    101
30-34           9.358974   73   73      73       73         73     73
35-39           5.256410   41   41      41       41         41     41
40+             1.666667   13   13      13       13         13     13

In [25]:
#reformat numbers to percentages
age_summary_df["Purchase ID"] = age_summary_df["Purchase ID"].map("{:,.2f}%".format)
age_summary_df

Purchase ID   SN  Age  Gender  Item ID  Item Name  Price
Age Summary                                                         
<10               2.95%   23   23      23       23         23     23
10-14             3.59%   28   28      28       28         28     28
15-19            17.44%  136  136     136      136        136    136
20-24            46.79%  365  365     365      365        365    365
25-29            12.95%  101  101     101      101        101    101
30-34             9.36%   73   73      73       73         73     73
35-39             5.26%   41   41      41       41         41     41
40+               1.67%   13   13      13       13         13     13

In [26]:
#reformat the table to only include columns of Age Summary, Purchase ID and write into DataFrame
org_age_summary_df = age_summary_df[["Purchase ID", "SN"]]
org_age_summary_df

Purchase ID   SN
Age Summary                 
<10               2.95%   23
10-14             3.59%   28
15-19            17.44%  136
20-24            46.79%  365
25-29            12.95%  101
30-34             9.36%   73
35-39             5.26%   41
40+               1.67%   13

In [27]:
#rename the columns for age demographics using the .rename command
final_group_summary_df = org_age_summary_df.rename(columns={"Purchase ID":"Percentage of Players", "SN":"Total Count"})
final_group_summary_df

Percentage of Players  Total Count
Age Summary                                   
<10                         2.95%           23
10-14                       3.59%           28
15-19                      17.44%          136
20-24                      46.79%          365
25-29                      12.95%          101
30-34                       9.36%           73
35-39                       5.26%           41
40+                         1.67%           13

### TOP SPENDORS ANALYSIS

- Run all the basic calculations to create results in a table


- Create a DataFrame for the summary of all results


- Create a cleaner format DataFrame for the summary

In [28]:
#group the purchase data by player spendors("SN")
original_purchased_data_df = pd.DataFrame(purchase_game)
original_purchased_data_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Summary  
0  Extraction, Quickblade Of Trembling Hands   3.53       20-24  
1                          Frenzied Scimitar   1.56         40+  
2                               Final Critic   4.88       20-24  
3                                Blindscythe   3.27       20-24  
4                                       Fury   1.44       20-24

In [29]:
#group the "SN"
group_SN_top_df = original_purchased_data_df.groupby("SN")
group_SN_top_df.count()

Purchase ID  Age  Gender  Item ID  Item Name  Price  \
SN                                                                   
Adairialis76             1    1       1        1          1      1   
Adastirin33              1    1       1        1          1      1   
Aeda94                   1    1       1        1          1      1   
Aela59                   1    1       1        1          1      1   
Aelaria33                1    1       1        1          1      1   
...                    ...  ...     ...      ...        ...    ...   
Yathecal82               3    3       3        3          3      3   
Yathedeu43               2    2       2        2          2      2   
Yoishirrala98            1    1       1        1          1      1   
Zhisrisu83               2    2       2        2          2      2   
Zontibe81                3    3       3        3          3      3   

               Age Summary  
SN                          
Adairialis76             1  
Adastirin33              1  
Aeda94                   1  
Aela59                   1  
Aelaria33                1  
...                    ...  
Yathecal82               3  
Yathedeu43               2  
Yoishirrala98            1  
Zhisrisu83               2  
Zontibe81                3  

[576 rows x 7 columns]

In [30]:
#start to work with the data created by new DataFrame
analysis_by_spendor_df = pd.DataFrame(group_SN_top_df["Purchase ID"].count())
analysis_by_spendor_df

Purchase ID
SN                        
Adairialis76             1
Adastirin33              1
Aeda94                   1
Aela59                   1
Aelaria33                1
...                    ...
Yathecal82               3
Yathedeu43               2
Yoishirrala98            1
Zhisrisu83               2
Zontibe81                3

[576 rows x 1 columns]

In [31]:
#get the total purchase value by SN
total_value_purchase_SN = group_SN_top_df["Price"].sum()
total_value_purchase_SN
dlr_total_value_purchase_SN = total_value_purchase_SN.map("${:,.2f}".format)
dlr_total_value_purchase_SN

SN
Adairialis76     $2.28
Adastirin33      $4.48
Aeda94           $4.91
Aela59           $4.32
Aelaria33        $1.79
                 ...  
Yathecal82       $6.22
Yathedeu43       $6.02
Yoishirrala98    $4.58
Zhisrisu83       $7.89
Zontibe81        $8.03
Name: Price, Length: 576, dtype: object

In [32]:
#organize the top sender data summary and get all the columns into an organized DataFrame
average_purchase_SN = group_SN_top_df["Price"].mean()
average_purchase_SN
dlr_average_purchase_SN = average_purchase_SN.map("${:,.2f}".format)
dlr_average_purchase_SN

SN
Adairialis76     $2.28
Adastirin33      $4.48
Aeda94           $4.91
Aela59           $4.32
Aelaria33        $1.79
                 ...  
Yathecal82       $2.07
Yathedeu43       $3.01
Yoishirrala98    $4.58
Zhisrisu83       $3.94
Zontibe81        $2.68
Name: Price, Length: 576, dtype: object

In [33]:
#organize the summary Top Spender data and get all columns to organize a new DataFrame with additional columns
analysis_by_spendor_df["Average Purchase Price"] = dlr_average_purchase_SN
analysis_by_spendor_df["Total Purchase Value"] = dlr_total_value_purchase_SN
analysis_by_spendor_df

Purchase ID Average Purchase Price Total Purchase Value
SN                                                                    
Adairialis76             1                  $2.28                $2.28
Adastirin33              1                  $4.48                $4.48
Aeda94                   1                  $4.91                $4.91
Aela59                   1                  $4.32                $4.32
Aelaria33                1                  $1.79                $1.79
...                    ...                    ...                  ...
Yathecal82               3                  $2.07                $6.22
Yathedeu43               2                  $3.01                $6.02
Yoishirrala98            1                  $4.58                $4.58
Zhisrisu83               2                  $3.94                $7.89
Zontibe81                3                  $2.68                $8.03

[576 rows x 3 columns]

In [34]:
#summary of the top spendor analysis and grouped by SN and rename the columns
summary_SN_purchased_data_df = analysis_by_spendor_df.rename(columns={"Purchase ID":"Purchase Count"})
top5_spendors_df = summary_SN_purchased_data_df.sort_values("Total Purchase Value", ascending=False)
top5_spendors_df.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                      
Haillyrgue51               3                  $3.17                $9.50
Phistym51                  2                  $4.75                $9.50
Lamil79                    2                  $4.64                $9.29
Aina42                     3                  $3.07                $9.22
Saesrideu94                2                  $4.59                $9.18

In [35]:
#group by item id and item name
group_top_item_df = original_purchased_data_df.groupby(["Item ID", "Item Name"])
group_top_item_df.count()

Purchase ID  SN  Age  \
Item ID Item Name                                                            
0       Splinter                                                4   4    4   
1       Crucifer                                                4   4    4   
2       Verdict                                                 6   6    6   
3       Phantomlight                                            6   6    6   
4       Bloodlord's Fetish                                      5   5    5   
...                                                           ...  ..  ...   
178     Oathbreaker, Last Hope of the Breaking Storm           12  12   12   
179     Wolf, Promise of the Moonwalker                         6   6    6   
181     Reaper's Toll                                           5   5    5   
182     Toothpick                                               3   3    3   
183     Dragon's Greatsword                                     3   3    3   

                                                      Gender  Price  \
Item ID Item Name                                                     
0       Splinter                                           4      4   
1       Crucifer                                           4      4   
2       Verdict                                            6      6   
3       Phantomlight                                       6      6   
4       Bloodlord's Fetish                                 5      5   
...                                                      ...    ...   
178     Oathbreaker, Last Hope of the Breaking Storm      12     12   
179     Wolf, Promise of the Moonwalker                    6      6   
181     Reaper's Toll                                      5      5   
182     Toothpick                                          3      3   
183     Dragon's Greatsword                                3      3   

                                                      Age Summary  
Item ID Item Name                                                  
0       Splinter                                                4  
1       Crucifer                                                4  
2       Verdict                                                 6  
3       Phantomlight                                            6  
4       Bloodlord's Fetish                                      5  
...                                                           ...  
178     Oathbreaker, Last Hope of the Breaking Storm           12  
179     Wolf, Promise of the Moonwalker                         6  
181     Reaper's Toll                                           5  
182     Toothpick                                               3  
183     Dragon's Greatsword                                     3  

[179 rows x 6 columns]

### MOST PROFITABLE ITEMS

- Sort the table from previous line that was created of the total purchase value in descending order

- Give the data a cleaner formatting

- Display a preview of the newly created DataFrame

In [36]:
#work with new data in DataFrame
analysis_by_item_df = pd.DataFrame(group_top_item_df["Purchase ID"].count())
analysis_by_item_df

Purchase ID
Item ID Item Name                                                
0       Splinter                                                4
1       Crucifer                                                4
2       Verdict                                                 6
3       Phantomlight                                            6
4       Bloodlord's Fetish                                      5
...                                                           ...
178     Oathbreaker, Last Hope of the Breaking Storm           12
179     Wolf, Promise of the Moonwalker                         6
181     Reaper's Toll                                           5
182     Toothpick                                               3
183     Dragon's Greatsword                                     3

[179 rows x 1 columns]

In [37]:
#calculate the new total purchase value by item
total_item_purchase_value = group_top_item_df["Price"].sum()
total_item_purchase_value
dlr_total_item_purchase_value = total_item_purchase_value.map("${:,.2f}".format)
dlr_total_item_purchase_value

Item ID  Item Name                                   
0        Splinter                                         $5.12
1        Crucifer                                        $11.77
2        Verdict                                         $14.88
3        Phantomlight                                    $14.94
4        Bloodlord's Fetish                               $8.50
                                                          ...  
178      Oathbreaker, Last Hope of the Breaking Storm    $50.76
179      Wolf, Promise of the Moonwalker                 $26.88
181      Reaper's Toll                                    $8.30
182      Toothpick                                       $12.09
183      Dragon's Greatsword                              $3.27
Name: Price, Length: 179, dtype: object

In [38]:
#calculate the purchase price by item
item_purchase_price = group_top_item_df["Price"].mean()
item_purchase_price
dlr_item_purchase_price = item_purchase_price.map("${:,.2f}".format)
dlr_item_purchase_price

Item ID  Item Name                                   
0        Splinter                                        $1.28
1        Crucifer                                        $2.94
2        Verdict                                         $2.48
3        Phantomlight                                    $2.49
4        Bloodlord's Fetish                              $1.70
                                                         ...  
178      Oathbreaker, Last Hope of the Breaking Storm    $4.23
179      Wolf, Promise of the Moonwalker                 $4.48
181      Reaper's Toll                                   $1.66
182      Toothpick                                       $4.03
183      Dragon's Greatsword                             $1.09
Name: Price, Length: 179, dtype: object

In [39]:
#re organize the item data into a summary and create columns for a new DataFrame
analysis_by_item_df["Item Price"] = dlr_item_purchase_price
analysis_by_item_df["Total Purchase Value"] = dlr_total_item_purchase_value
analysis_by_item_df

Purchase ID Item Price  \
Item ID Item Name                                                              
0       Splinter                                                4      $1.28   
1       Crucifer                                                4      $2.94   
2       Verdict                                                 6      $2.48   
3       Phantomlight                                            6      $2.49   
4       Bloodlord's Fetish                                      5      $1.70   
...                                                           ...        ...   
178     Oathbreaker, Last Hope of the Breaking Storm           12      $4.23   
179     Wolf, Promise of the Moonwalker                         6      $4.48   
181     Reaper's Toll                                           5      $1.66   
182     Toothpick                                               3      $4.03   
183     Dragon's Greatsword                                     3      $1.09   

                                                     Total Purchase Value  
Item ID Item Name                                                          
0       Splinter                                                    $5.12  
1       Crucifer                                                   $11.77  
2       Verdict                                                    $14.88  
3       Phantomlight                                               $14.94  
4       Bloodlord's Fetish                                          $8.50  
...                                                                   ...  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
179     Wolf, Promise of the Moonwalker                            $26.88  
181     Reaper's Toll                                               $8.30  
182     Toothpick                                                  $12.09  
183     Dragon's Greatsword                                         $3.27  

[179 rows x 3 columns]

In [40]:
#create a summary of most popular item analysis grouped by item
item_sum_purchased_data_df = analysis_by_item_df.rename(columns={"Purchase ID":"Purchase Count"})
top5_items_df = item_sum_purchased_data_df.sort_values("Purchase Count", ascending=False)
top5_items_df.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [41]:
#create a total purchase price column in number form
analysis_by_item_df["Total Purchase Value"] = total_item_purchase_value
analysis_by_item_df

Purchase ID Item Price  \
Item ID Item Name                                                              
0       Splinter                                                4      $1.28   
1       Crucifer                                                4      $2.94   
2       Verdict                                                 6      $2.48   
3       Phantomlight                                            6      $2.49   
4       Bloodlord's Fetish                                      5      $1.70   
...                                                           ...        ...   
178     Oathbreaker, Last Hope of the Breaking Storm           12      $4.23   
179     Wolf, Promise of the Moonwalker                         6      $4.48   
181     Reaper's Toll                                           5      $1.66   
182     Toothpick                                               3      $4.03   
183     Dragon's Greatsword                                     3      $1.09   

                                                      Total Purchase Value  
Item ID Item Name                                                           
0       Splinter                                                      5.12  
1       Crucifer                                                     11.77  
2       Verdict                                                      14.88  
3       Phantomlight                                                 14.94  
4       Bloodlord's Fetish                                            8.50  
...                                                                    ...  
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
179     Wolf, Promise of the Moonwalker                              26.88  
181     Reaper's Toll                                                 8.30  
182     Toothpick                                                    12.09  
183     Dragon's Greatsword                                           3.27  

[179 rows x 3 columns]

In [42]:
#have an analysis grouped by item and rename the Purchase ID column
top5_items_df=item_sum_purchased_data_df.sort_values("Total Purchase Value", ascending=False)


#format the total purchase price to be in dollar form
dlr_total_item_purchase_value = total_item_purchase_value.map("${:,.2f}".format)
top5_items_df["Total Purchase Value"] = dlr_total_item_purchase_value
top5_items_df.head()

Purchase Count Item Price  \
Item ID Item Name                                                      
63      Stormfury Mace                                  2      $4.99   
29      Chaos, Ender of the End                         5      $1.98   
173     Stormfury Longsword                             2      $4.93   
38      The Void, Vengeance of Dark Magic               4      $2.37   
143     Frenzied Scimitar                               6      $1.56   

                                          Total Purchase Value  
Item ID Item Name                                               
63      Stormfury Mace                                   $9.98  
29      Chaos, Ender of the End                          $9.90  
173     Stormfury Longsword                              $9.86  
38      The Void, Vengeance of Dark Magic                $9.48  
143     Frenzied Scimitar                                $9.36

### CONCLUSIONS:

- The largest part of the active player populations are males that take around 84% of the population while the female players take up around 14%.


- The peak age demographic that was determined through the DataFrame fell between the ages of 20-24 years old which took up ~47% of all demographic groups. Right behind that group was the second age demographic of 15-19 year olds which took up around 17% of the age demographic.

- There were 183 unique items offered in the game; the most popular ones were Oathbreaker, Last Hope of the Breaking Storm which sent in $51 to the game, Nirvana and Fiery Glass Crusader bringing $44 and $41 each. All of the 780 players preferred different items and there was no large significance between the players and items.


- The average puchase a player made in the game was around $3 and the top spendors paid three times as much  - around $9 for in-game purchases. The total profit from all of the sold items was around $2400 for the total players.